packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -268,47 +269,376 @@ class _CupertinoAppState extends State<CupertinoApp> {
+              onUnknownRoute: widget.onUnknownRoute,
+              builder: widget.builder,
+              title: widget.title,
+              onGenerateTitle: widget.onGenerateTitle,
+              textStyle: effectiveThemeData.textTheme.textStyle,
+              color: widget.color ?? CupertinoColors.activeBlue,
+              locale: widget.locale,
+              localizationsDelegates: _localizationsDelegates,
+              localeResolutionCallback: widget.localeResolutionCallback,
+              localeListResolutionCallback: widget.localeListResolutionCallback,
+              supportedLocales: widget.supportedLocales,
+              showPerformanceOverlay: widget.showPerformanceOverlay,
+              checkerboardRasterCacheImages: widget.checkerboardRasterCacheImages,
+              checkerboardOffscreenLayers: widget.checkerboardOffscreenLayers,
+              showSemanticsDebugger: widget.showSemanticsDebugger,
+              debugShowCheckedModeBanner: widget.debugShowCheckedModeBanner,
+              inspectorSelectButtonBuilder: (BuildContext context, VoidCallback onPressed) {
+                return CupertinoButton.filled(
+                  child: const Icon(
+                    CupertinoIcons.search,
+                    size: 28.0,
+                    color: CupertinoColors.white,
+                  ),
+                  padding: EdgeInsets.zero,
+                  onPressed: onPressed,
+                );
+              },
+            ),
+          ),
         ),
       ),
     );
   }
 }
+
+/// Establishes a subtree where iOS system colors resolve to the given data.
+///
+/// Typically the given [CupertinoSystemColorsData] is resolved against its own
+/// [BuildContext] using [CupertinoSystemColorsData.resolveColors].
+class CupertinoSystemColors extends InheritedWidget {
+  const CupertinoSystemColors._({
+    Key key,
+    @required CupertinoSystemColorsData data,
+    Widget child,
+  }) : _data = data,
+       assert(data != null),
+       super(key: key, child: child);
+
+  /// Creates a widget that provides a [CupertinoSystemColorsData] which is extracted
+  /// from the given [BuildContext] to its descendants.
+  ///
+  /// The [context] and [child] arguments must not be null.
+  CupertinoSystemColors.fromCurrentContext({
+    Key key,
+    BuildContext context,
+    Widget child,
+  }) : this._(key: key, data: CupertinoSystemColors.of(context).resolveColors(context), child: child);
+
+  final CupertinoSystemColorsData _data;
+
+  @override
+  bool updateShouldNotify(CupertinoSystemColors oldWidget) => oldWidget._data != _data;
+
+
+  /// Retrieves the iOS system colors from the given [BuildContext].
+  ///
+  /// Falls back to [fromSystem] and then [fallbackValues], if such information
+  /// is not available in the given [BuildContext].
+  static CupertinoSystemColorsData of(BuildContext context) {
+    final CupertinoSystemColors widget = context.inheritFromWidgetOfExactType(CupertinoSystemColors);
```

Comment:
I think this would be simpler if there was a nullOk parameter and, if nullOk was false, it returned _fallbackValues. Don't think fallback values needs to be public.

Reply:
The optional parameter is usually called `nullOk`.

Reply:
But it's doing something a bit different from regular `nullOk`, it doesn't throw an exception even when `nullOk` is false. Also since `useFallbackValues` is now marked as `visibleForTesting`, people won't be able to do `CupertinoSystemColorsData.of(context, nullOk: true) ?? CupertinoSystemColors.fallbackValues`

Reply:
That sounds OK to me.